dipy/segment/metric.py

```diff
@@ -39,3 +40,49 @@ def mdf(s1, s2):
                         vol 6, no 175, 2012.
     """
     return dist(MinimumAverageDirectFlipMetric(), s1, s2)
+
+
+def mean_l1_func(a, b):
+    """ Compute the average L1 distance (MDF without flip)
+
+    Arrays are representing a single streamline or a list of streamlines
+    that have the same number of N-dimensional points (two last axis).
+
+    Parameters
+    ----------
+    a : 2D or 3D array
+        A streamline or concatenated streamlines
+        (array of S streamlines by P points in N dimension).
+    b : 2D or 3D array
+        A streamline or concatenated streamlines
+        (array of S streamlines by P points in N dimension).
+
+    Returns
+    -------
+    1D array
+        Distance between each S streamlines
+    """
+    return np.mean(np.sum(np.abs(a - b), axis=-1), axis=-1)
+
+
+def mean_l2_func(a, b):
```

Comment:
I could rename it to mean_euclidean_dist()

scipy.spatial.distance.euclidean was made for for comparing 2 vectors (not 2 array of vector)
I could use "linalg norm 2" however we would have to reshape the list of streamlines 
in a list of points and reshape it back for the average.

Overall this is a little bit faster (no check, not a general exponent) otherwise it would look like this
```
diff = a - b
dists = np.linalg.norm(diff.reshape((-1, 3)), axis=-1))
return np.mean(dists.reshape((len(diff), -1)), axis=-1))
```

Reply:
it was renamed